pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1728,15 +1736,114 @@ private boolean isTmpAndCanDelete(String filePath, Set<String> downloadUrls, Pin
    * @return the set of consuming segments for which commit was initiated
    */
   public Set<String> forceCommit(String tableNameWithType, @Nullable String partitionGroupIdsToCommit,
-      @Nullable String segmentsToCommit) {
+      @Nullable String segmentsToCommit, int batchSize) {
     IdealState idealState = getIdealState(tableNameWithType);
     Set<String> allConsumingSegments = findConsumingSegments(idealState);
     Set<String> targetConsumingSegments = filterSegmentsToCommit(allConsumingSegments, partitionGroupIdsToCommit,
         segmentsToCommit);
-    sendForceCommitMessageToServers(tableNameWithType, targetConsumingSegments);
+
+    List<Set<String>> segmentBatchList = getSegmentBatchList(idealState, targetConsumingSegments, batchSize);
+    ExecutorService executorService = Executors.newFixedThreadPool(1);
+
+    try {
+      for (Set<String> segmentBatchToCommit : segmentBatchList) {
+        executorService.submit(() -> executeBatch(tableNameWithType, segmentBatchToCommit));
+      }
+    } finally {
+      executorService.shutdown();
+    }
+
     return targetConsumingSegments;
   }
 
+  private void executeBatch(String tableNameWithType, Set<String> segmentBatchToCommit) {
+    sendForceCommitMessageToServers(tableNameWithType, segmentBatchToCommit);
+
+    int attemptCount = 0;
+    try {
+      attemptCount = DEFAULT_RETRY_POLICY.attempt(() -> isBatchSuccessful(tableNameWithType, segmentBatchToCommit));
+    } catch (AttemptsExceededException | RetriableOperationException e) {
+      String errorMsg =
+          String.format("Failed to execute the forceCommit batch of segments: %s , attempt count: %d",
```

Comment:
We want to log the individual segments not successfully committed. Currently there is no way to differentiate the ones committed and ones failed